### ISSUE TITLE
`ray` version causing hung state in best_quality mode

### ISSUE THREAD

**Issue by nealgravindra** [autogluon/autogluon#4300]
When running AutoGluon in best_quality mode, I encountered a hung state with the following error:

```
(autoscaler +51m31s) Error: No available node types can fulfill resource request {'CPU': 72.0}. Add suitable node types to this cluster to resolve this issue.
```

The issue was resolved by uninstalling and reinstalling`ray`:
```
pip uninstall ray
pip install ray
```

It appears that AutoGluon requires 2.10.0 <= ray < 2.11.0, but installing ray 2.31.0 resolves this issue for my machine. 

As such, I thought it made sense for a suggestion, namely, to consider updating the requirements to allow for new versions of `ray`

**Comment by nvidia-wtruong** [comment]
Thanks for raising this issue - a simple reinstall helped me :)

**Comment by arturdaraujo** [comment]
I'm getting this "Will use sequential fold fitting strategy because import of ray failed. Reason: ray==2.31.0 detected. 2.10.0 <= ray < 2.11.0 is required. You can use pip to install certain version of ray `pip install ray==2.10.0` ".

Verbosity: 2 (Standard Logging)
=================== System Info ===================
AutoGluon Version:  1.1.1
Python Version:     3.11.9
Operating System:   Linux
Platform Machine:   x86_64
Platform Version:   #1 SMP Fri Mar 29 23:14:13 UTC 2024
CPU Count:          8
Memory Avail:       12.22 GB / 23.47 GB (52.1%)
Disk Space Avail:   362.31 GB / 911.84 GB (39.7%)
===================================================
Presets specified: ['medium_quality']
Warning: Training may take a very long time because `time_limit` was not specified and `train_data` is large (4500000 samples, 918.0 MB).
        Consider setting `time_limit` to ensure training finishes within an expected duration or experiment with a small portion of `train_data` to identify an ideal `presets` and `hyperparameters` configuration.
Beginning AutoGluon training ...
AutoGluon will save models to "/mnt/d/python_directory_2/models_temp"
Train Data Rows:    4500000
Train Data Columns: 47
Tuning Data Rows:    500000
Tuning Data Columns: 47
Label Column:       target
AutoGluon infers your prediction problem is: 'binary' (because only two unique label-values observed).
        2 unique label values:  [1, 0]
        If 'binary' is not the correct problem_type, please manually specify the problem_type parameter during Predictor init (You may specify problem_type as one of: ['binary', 'multiclass', 'regression', 'quantile'])
Problem Type:       binary
Preprocessing data ...
Selected class <--> label mapping:  class 1 = 1, class 0 = 0
Using Feature Generators to preprocess the data ...
Fitting AutoMLPipelineFeatureGenerator...
        Available Memory:                    11608.47 MB
        Train Data (Original)  Memory Usage: 896.45 MB (7.7% of available memory)
        Warning: Data size prior to feature transformation consumes 7.7% of available memory. Consider increasing memory or subsampling the data to avoid instability.
        Inferring data type of each feature based on column values. Set feature_metadata_in to manually specify special dtypes of the features.
        Stage 1 Generators:
                Fitting AsTypeFeatureGenerator...
                        Note: Converting 1 features to boolean dtype as they only contain 2 unique values.
        Stage 2 Generators:
                Fitting FillNaFeatureGenerator...
        Stage 3 Generators:
                Fitting IdentityFeatureGenerator...
        Stage 4 Generators:
                Fitting DropUniqueFeatureGenerator...
        Stage 5 Generators:
                Fitting DropDuplicatesFeatureGenerator...
        Useless Original Features (Count: 5): ['ASIN_30_n_03s__FC_LocalSimple_mean1_tauresrat', 'COSH_15_n_03m__SC_FluctAnal_2_dfa_50_1_2_logi_prop_r1', 'ROC_30_n_15m__SC_FluctAnal_2_rsrangefit_50_1_logi_prop_r1', 'TANH_30_n_15m__CO_FirstMin_ac', 'NATR_5_n_30m__SC_FluctAnal_2_rsrangefit_50_1_logi_prop_r1']
                These features carry no predictive signal and should be manually investigated.
                This is typically a feature which has the same value for all rows.
                These features do not need to be present at inference time.
        Types of features in original data (raw dtype, special dtypes):
                ('float', []) : 42 | ['MACDEXT_macdsignal_5_n_01s__DN_OutlierInclude_n_001_mdrmd', 'DEMA_10_n_03s__SB_TransitionMatrix_3ac_sumdiagcov', 'MACDEXT_macd_3_n_03s__SB_BinaryStats_mean_longstretch1', 'STOCH_slowd_45_n_03s__IN_AutoMutualInfoStats_40_gaussian_fmmi', 'ULTOSC_45_n_03s__DN_Spread_Std', ...]
        Types of features in processed data (raw dtype, special dtypes):
                ('float', [])     : 41 | ['MACDEXT_macdsignal_5_n_01s__DN_OutlierInclude_n_001_mdrmd', 'DEMA_10_n_03s__SB_TransitionMatrix_3ac_sumdiagcov', 'STOCH_slowd_45_n_03s__IN_AutoMutualInfoStats_40_gaussian_fmmi', 'ULTOSC_45_n_03s__DN_Spread_Std', 'AROONOSC_5_n_05s__SP_Summaries_welch_rect_area_5_1', ...]
                ('int', ['bool']) :  1 | ['MACDEXT_macd_3_n_03s__SB_BinaryStats_mean_longstretch1']
        10.5s = Fit runtime
        42 features in original data used to generate 42 features in processed data.
        Train Data (Processed) Memory Usage: 786.78 MB (6.8% of available memory)
Data preprocessing and feature engineering runtime = 11.56s ...
AutoGluon will gauge predictive performance using evaluation metric: 'roc_auc'
        This metric expects predicted probabilities rather than predicted class labels, so you'll need to use predict_proba() instead of predict()
        To change this, specify the eval_metric parameter of Predictor()
use_bag_holdout=True, will use tuning_data as holdout (will not be used for early stopping).
User-specified model hyperparameters to be fit:
{
        'GBM': [{}, 'GBMLarge'],
        'RF': [{'criterion': 'gini', 'ag_args': {'name_suffix': 'Gini', 'problem_types': ['binary', 'multiclass']}}, {'criterion': 'entropy', 'ag_args': {'name_suffix': 'Entr', 'problem_types': ['binary', 'multiclass']}}, {'criterion': 'mse', 'ag_args': {'name_suffix': 'MSE', 'problem_types': ['regression']}}],
        'XT': [{'criterion': 'gini', 'ag_args': {'name_suffix': 'Gini', 'problem_types': ['binary', 'multiclass']}}, {'criterion': 'entropy', 'ag_args': {'name_suffix': 'Entr', 'problem_types': ['binary', 'multiclass']}}, {'criterion': 'mse', 'ag_args': {'name_suffix': 'MSE', 'problem_types': ['regression']}}],
        'KNN': [{'weights': 'uniform', 'ag_args': {'name_suffix': 'Unif'}}, {'weights': 'distance', 'ag_args': {'name_suffix': 'Dist'}}],
}
Excluded models: ['RF', 'KNN', 'XT'] (Specified by `excluded_model_types`)
Fitting 2 L1 models ...
Fitting model: LightGBM_BAG_L1 ...
Will use sequential fold fitting strategy because import of ray failed. Reason: ray==2.31.0 detected. 2.10.0 <= ray < 2.11.0 is required. You can use pip to install certain version of ray `pip install ray==2.10.0`

**Comment by Innixma** [comment]
@jtele2 AutoGluon will use the latest ray version in the v1.2 release coming later this month (November 2024).

**Comment by nealgravindra** [comment]
Awesome, thanks! And specifying the version did help. Cheers!
